File: codegen/results/v002/ollama/qwen3.5_35b-a3b-coding-nvfp4_nothink/20260414_100330/python/007_beatles_interview/solution.py
import csv
import json
from datetime import datetime
from dateutil.relativedelta import relativedelta

def parse_date(date_str):
    if not date_str or date_str.lower() == 'null':
        return None
    # Handle various date formats: MM/DD/YYYY or YYYY-MM-DD
    try:
        return datetime.strptime(date_str, "%m/%d/%Y")
    except ValueError:
        try:
            return datetime.strptime(date_str, "%Y-%m-%d")
        except ValueError:
            return None

def calculate_age(birthday_str):
    birthday = parse_date(birthday_str)
    if not birthday:
        return 0
    
    reference_date = datetime(2025, 7, 1)
    age = reference_date.year - birthday.year
    if (reference_date.month, reference_date.day) < (birthday.month, birthday.day):
        age -= 1
    return age

def process_person(row):
    # Parse Name: "First Last" or "First Middle Last" -> First Last
    name_parts = row['Name'].strip().split()
    first_name = name_parts[0]
    last_name = name_parts[-1] # Take the last part as surname
    
    birthday = row['Birthday']
    age = calculate_age(birthday)
    
    relatives = []
    
    father_name = row['Father']
    if father_name and father_name.lower() != 'null':
        parts = father_name.strip().split()
        relatives.append({
            "FirstName": parts[0],
            "LastName": parts[-1],
            "Relationship": "Father"
        })
        
    mother_name = row['Mother']
    if mother_name and mother_name.lower() != 'null':
        parts = mother_name.strip().split()
        relatives.append({
            "FirstName": parts[0],
            "LastName": parts[-1],
            "Relationship": "Mother"
        })
        
    brother_name = row['Brother']
    if brother_name and brother_name.lower() != 'null':
        parts = brother_name.strip().split()
        relatives.append({
            "FirstName": parts[0],
            "LastName": parts[-1],
            "Relationship": "Brother"
        })
        
    sister_name = row['Sister']
    if sister_name and sister_name.lower() != 'null':
        parts = sister_name.strip().split()
        relatives.append({
            "FirstName": parts[0],
            "LastName": parts[-1],
            "Relationship": "Sister"
        })
        
    return {
        "FirstName": first_name,
        "LastName": last_name,
        "Birthday": birthday,
        "Age": age,
        "Relatives": relatives
    }

def main():
    results = []
    
    with open('input/input.csv', 'r', newline='', encoding='utf-8') as csvfile:
        reader = csv.DictReader(csvfile)
        for row in reader:
            person = process_person(row)
            # Format birthday as ISO string
            person['Birthday'] = person['Birthday'].strftime('%Y-%m-%d')
            results.append(person)
            
    print(json.dumps(results, indent=2))

if __name__ == "__main__":
    main()

File: codegen/results/v002/ollama/qwen3.5_35b-a3b-coding-nvfp4_nothink/20260414_100330/typescript/007_beatles_interview/solution.ts
import * as fs from 'fs';

const inputPath = 'input/input.csv';
const outputPath = 'input/expected_format.json';

// Read the input CSV
const csvContent = fs.readFileSync(inputPath, 'utf-8');
const lines = csvContent.trim().split('\n');

// Skip header
const dataLines = lines.slice(1);

// Reference date for age calculation
const refDate = new Date('2025-07-01');

function calculateAge(birthdayStr: string): number {
  // Parse various date formats: MM/DD/YYYY or M/D/YYYY
  const parts = birthdayStr.split('/');
  const month = parseInt(parts[0], 10);
  const day = parseInt(parts[1], 10);
  const year = parseInt(parts[2], 10);

  const birthDate = new Date(year, month - 1, day);

  let age = refDate.getFullYear() - birthDate.getFullYear();
  const monthDiff = refDate.getMonth() - birthDate.getMonth();
  if (monthDiff < 0 || (monthDiff === 0 && refDate.getDate() < birthDate.getDate())) {
    age--;
  }
  return age;
}

function parseName(name: string): { firstName: string; lastName: string } {
  const parts = name.trim().split(' ');
  if (parts.length >= 2) {
    const lastName = parts.pop()!;
    const firstName = parts.join(' ');
    return { firstName, lastName };
  }
  return { firstName: name.trim(), lastName: '' };
}

function parseRelative(name: string, relationship: string): { firstName: string; lastName: string; relationship: string } | null {
  if (!name || name === 'null') return null;
  const { firstName, lastName } = parseName(name);
  return { firstName, lastName, relationship };
}

const result: any[] = [];

for (const line of dataLines) {
  const parts = line.split(',');
  if (parts.length < 2) continue;

  const name = parts[0];
  const birthday = parts[1];
  const died = parts[2]; // Not used in output, but present in CSV
  const fatherName = parts[3];
  const motherName = parts[4];
  const brotherName = parts[5];
  const sisterName = parts[6];

  const { firstName, lastName } = parseName(name);
  const age = calculateAge(birthday);

  const relatives: { firstName: string; lastName: string; relationship: string }[] = [];

  const father = parseRelative(fatherName, 'Father');
  if (father) relatives.push(father);

  const mother = parseRelative(motherName, 'Mother');
  if (mother) relatives.push(mother);

  const brother = parseRelative(brotherName, 'Brother');
  if (brother) relatives.push(brother);

  const sister = parseRelative(sisterName, 'Sister');
  if (sister) relatives.push(sister);

  result.push({
    FirstName: firstName,
    LastName: lastName,
    Birthday: birthday,
    Age: age,
    Relatives: relatives
  });
}

console.log(JSON.stringify(result, null, 2));